lib/ResolverFactory.js

```diff
@@ -434,21 +439,23 @@ function aliasOptionsToArray(alias) {
  * Merging filtered elements
  * @param {string[]} array source array
  * @param {function(string): boolean} filter predicate
- * @returns {Array<string | string[]>} merge result
+ * @returns {Set<string | string[]>} merge result
  */
-function mergeFilteredToArray(array, filter) {
-	return array.reduce((/** @type {Array<string | string[]>} */ array, item) => {
-		if (filter(item)) {
-			const lastElement = /** @type {string[]} */ (array[array.length - 1]);
-			if (Array.isArray(lastElement)) {
-				lastElement.push(item);
+function mergeFilteredToSet(array, filter) {
+	return new Set(
+		array.reduce((/** @type {Array<string | string[]>} */ array, item) => {
```

Comment:
Set need to be applied to the incoming `array`. Due to the nested structure a Set as result is not needed/has not the expected effect.